### PR TITLE
Issue 148 connection helper

### PR DESCRIPTION
**Pull request by vsuaste**
* Modify schema and models to handle direct access to nodes in type Connection queries.
* Unit test fixed
* Added integration test

### CODE REVIEW

**Comment by asishallab** [comment]
If you merge, please make sure that the following issues are also taken care of:
* Any other storage type, e.g. Cassandra, should work with this new helper, too
* In case of empty foreign key arrays, make sure an empty helper is returned, too.
* Write integration tests to ensure the above

File: .testenv.json [hunk 1]
```diff
@@ -8,13 +8,13 @@
     {
       "name": "gql_science_db_graphql_server1",
       "template": "https://github.com/Zendro-dev/graphql-server.git",
-      "branch": "master",
+      "branch": "issue-148-connection-helper",
```
Comment (wunderbarr) [inline .testenv.json:11]: change it to master

File: .testenv.json [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
       "url": "http://localhost:3000/graphql"
     },
     {
       "name": "gql_science_db_graphql_server2",
       "template": "https://github.com/Zendro-dev/graphql-server.git",
-      "branch": "master",
+      "branch": "issue-148-connection-helper",
```
Comment (wunderbarr) [inline .testenv.json:17]: change it to master

File: test/mocha_integration_test.js [hunk 3]
```diff
@@ -3117,4 +3117,73 @@ describe(
...
+
+  describe('Helper connection for direct acces to nodes', function() {
+    //set up the environment
+    before(async function(){
+      //measurements for sql and zendro-server tests
+      let res = itHelpers.request_graph_ql_post('mutation{addBook(id:"remote_b1" title:"t1"){id} }');
+      expect(res.statusCode).to.equal(200);
+      res = itHelpers.request_graph_ql_post('mutation{addBook(id:"remote_b2" title:"t2"){id } }');
+      expect(res.statusCode).to.equal(200);
+      res = itHelpers.request_graph_ql_post('mutation{addAuthor(id:"remote_a1" name:"n1" addBooks:["remote_b1","remote_b2"]){id }}');
```
Comment (wunderbarr) [inline test/mocha_integration_test.js:3130]: add booksConnection for the test

File: test/mocha_integration_test.js [hunk 4]
```diff
@@ -3117,4 +3117,73 @@ describe(
...
+      expect(res.statusCode).to.equal(200);
+
+      res = itHelpers.request_graph_ql_post_instance2('mutation{addSq_book(id:"remote_b1" title:"t1"){id} }');
+      expect(res.statusCode).to.equal(200);
+      res = itHelpers.request_graph_ql_post_instance2('mutation{addSq_book(id:"local_b1" title:"t1"){id} }');
+      expect(res.statusCode).to.equal(200);
+
+      res = itHelpers.request_graph_ql_post_instance2('mutation{addSq_author(id:"remote_a2" name:"n2" addBooks:["remote_b1"]){id}}');
+      expect(res.statusCode).to.equal(200);
+      res = itHelpers.request_graph_ql_post_instance2('mutation{addSq_author(id:"local_a2" name:"n2" addBooks:[ "local_b1"]){id}}');
```
Comment (wunderbarr) [inline test/mocha_integration_test.js:3142]: add bookConnection for the test

File: test/mocha_integration_test.js [hunk 5]
```diff
@@ -3117,4 +3117,73 @@ describe(
...
+    //set up the environment
+    before(async function(){
+      //measurements for sql and zendro-server tests
+      let res = itHelpers.request_graph_ql_post('mutation{addBook(id:"remote_b1" title:"t1"){id} }');
+      expect(res.statusCode).to.equal(200);
+      res = itHelpers.request_graph_ql_post('mutation{addBook(id:"remote_b2" title:"t2"){id } }');
+      expect(res.statusCode).to.equal(200);
+      res = itHelpers.request_graph_ql_post('mutation{addAuthor(id:"remote_a1" name:"n1" addBooks:["remote_b1","remote_b2"]){id }}');
+      expect(res.statusCode).to.equal(200);
+      res = itHelpers.request_graph_ql_post('mutation{addAuthor(id:"remote_a2" name:"n2" addBooks:["remote_b1"]){id}}');
```
Comment (wunderbarr) [inline test/mocha_integration_test.js:3132]: add booksConnection for the test

**Review by wunderbarr — CHANGES_REQUESTED** [review]
.

File: test/mocha_integration_test.js (hunk 5 above)
Comment (vsuaste) [reply test/mocha_integration_test.js:3132]: This is not the test, this is only the setup for the test, which is below.

File: test/mocha_integration_test.js (hunk 4 above)
Comment (vsuaste) [reply test/mocha_integration_test.js:3142]: Same observation as above.

**Pull request merged**